pkg/iac/terraform/state/alerts.go

```diff
@@ -0,0 +1,40 @@
+package state
+
+import "fmt"
+
+type StateReadAlert struct {
+	key string
+	err string
+}
+
+func NewStateReadAlert(key string, err error) *StateReadAlert {
+	return &StateReadAlert{key: key, err: err.Error()}
+}
+
+func (s *StateReadAlert) Message() string {
+	return fmt.Sprintf("Your analysis will be incomplete. There was an error reading state file '%s': %s", s.key, s.err)
+}
+
+func (s *StateReadAlert) ShouldIgnoreResource() bool {
+	return false
+}
+
+type StateEnumerationAlert struct {
+	source string
+	err    string
+}
+
+func NewStateEnumerationAlertAlert(err error) *StateEnumerationAlert {
```

Comment:
typo to fix

Reply:
> I still get a blocking error when the a specified file does not exist, is it supposed to happen ?
> 
> ```
> $ AWS_PROFILE=driftctlrole driftctl scan -f 'tfstate://../tf+sundowndev/*.tfstate' -f 'tfstate://terraform1.tfstate' --tf-provider-version 3.44.0
> DEBU[0000] Bound environment variable to flag env=disable_telemetry flag=disable-telemetry value=true
> DEBU[0000] New provider library created
> DEBU[0000] Found existing provider path=/home/sundowndev/.driftctl/plugins/linux_amd64/terraform-provider-aws_v3.44.0_x5
> DEBU[0000] Starting gRPC client alias=us-east-1
> DEBU[0002] New gRPC client started alias=us-east-1
> DEBU[0002] Terraform provider initialized alias=us-east-1 name=aws
> DEBU[0002] Checking for driftignore
> DEBU[0002] Found IAC supplier backend= path=../tf+sundowndev/*.tfstate supplier=tfstate
> DEBU[0002] Found IAC supplier backend= path=terraform1.tfstate supplier=tfstate
> INFO[0002] Start reading IaC
> DEBU[0002] Stopping ParallelRunner
> Scanned states (0) 
> Your analysis will be incomplete. There was an error reading state file 'terraform1.tfstate': lstat terraform1.tfstate: no such file or directory
> DEBU[0002] Enumerated keys keys=[../tf+sundowndev/terraform.tfstate]
> DEBU[0002] Reading resources from state backend= path=../tf+sundowndev/terraform.tfstate
> DEBU[0002] Progress received a tic after stopping. Restarting...
> DEBU[0002] Closing providers
> DEBU[0002] Closing provider key=aws
> DEBU[0002] Closing gRPC client alias=us-east-1
> DEBU[0002] Found module in state module= resourceCount=3
> DEBU[0002] Ignored resource from state since it is ignored in filter name=b type=aws_iam_policy
> lstat terraform1.tfstate: no such file or directory
> exit status 1
> ```

ok the PR was missing a commit where we actually use the IacChainSupplier :`)

Reply:
the typo is still there ?

Reply:
sorry forgot to remove it